enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,454 @@
+>
+> What trade-offs (technical/efficiency cost, user experience, flexibility,
+> supportability, etc) must be made in order to implement this? What are the reasons
+> we might not want to undertake this proposal, and how do we overcome them?
+>
+> Does this proposal implement a behavior that's new/unique/novel? Is it poorly
+> aligned with existing user expectations?  Will it be a significant maintenance
+> burden?  Is it likely to be superceded by something else in the near future?
+
+The key tradeoff is that we will start treating Gateway API effectively as if
+it were a core API. This means any perceived value someone might see in being
+able to manage and upgrade the Gateway API resources themselves, independently
+of the platform version, will be lost. Third party implementations will not be
+able to bring their own API changes.
+
+> **Note**: The API has stabilized, so it's important to note that there is
+> little expectation of any rapid change.
+
+> **Note**: While third party implementations can not bring their own API
+> changes directly, if the need for more flexibility arises we may consider
+> adding functionality in future iterations that enables the platform to more
+> dynamically set the version deployed in the cluster.
+
+> **Note**: This "downside" can actually be an upside for integrators, as we've
+> heard from projects which have Gateway API integrations (such as OSSM) which
+> appreciate having a consistent and known version of Gateway API on the
+> cluster which they can always rely on being there.
+
+Unfortunately this is not really an avoidable problem, as it is not tenable for
+us to simultaneously say that we support Gateway API as a primary and fully
+supported API surface for ingress traffic, and also have no control over which
+version of the APIs are present, or if they are even present at all (e.g. the
+cluster admin decides to delete them).
+
+To make this situation a bit easier we do anticipate providing some updates
+after the initial release. We want to eventually allow version _ranges_ in
+time after we resolve the "dead fields" problem (see more in the section about
+this above) which we expect to provide significantly more flexibility and take
+care of many concerns that would come from this change. We are also tracking and
+supporting [upstream efforts] to separate experimental APIs out into their own
+group, which will provide more flexibility when users want experimental features.
+
+[upstream efforts]:https://github.com/kubernetes-sigs/gateway-api/discussions/3497
+
+## Test Plan
+
+The Ingress Operator will have tests to validate the intended functionality
+outlined in this EP:
+
+- CIO Tests that verify the pre-upgrade check logic
+- CIO Tests that verify the post-upgrade logic:
+    - Delete the CRDs; verify that the operator re-installs them
+    - Install incompatible CRDs; verify that the operator reports `Degraded`
+    - Install experimental CRDs; verify that the operator reports `Degraded`
```

Comment:
Is this correct, or is the operator removing these now?

Reply:
This is correct, specifically this is for the tests: if the experimental APIs somehow become present we go degraded. We want to make experimental available in a later release, but for initial release we're being very strict about it not being present because it can cause some issues. We are considering a very deliberate way to enable Experimental later, where the cluster gets set to `TechPreviewNoUpgrade` and then the user can do what they like with Experimental, but that's expected to be a future enhancement proposal and for now :lock:.